public/app/features/canvas/element.ts

```diff
@@ -72,7 +72,7 @@ export interface CanvasElementItem<TConfig = any, TData = any> extends RegistryI
   /** The default width/height to use when adding  */
   defaultSize?: Placement;
 
-  prepareData?: (ctx: DimensionContext, cfg: TConfig) => TData;
+  prepareData?: (dimensionContext: DimensionContext, elementOptions: CanvasElementOptions<TConfig>) => TData;
```

Comment:
This is the primary change / refactor that is allowing the more universal data link support. TLDR; before we were just passing the element's config (specific options for each element) whereas now we are passing down the entire element's options which includes the specific config as well as the the generic container background / border info that can be tied to field data and thus be applicable to data link functionality